### PR TITLE
[WIP^2] Introducing CalDAV into core

### PR DESCRIPTION
**Pull request by DeepDiver1975**
To avoid conflicts we first have to get https://github.com/owncloud/core/pull/20184/files in.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
WTF?

````
4. put_get_utf8_segment.. [Wed Nov  4 14:33:06 2015] PHP Fatal error:  Access to undeclared static property: OC\Files\Filesystem::$normalizedPathCache in /home/travis/build/owncloud/core/lib/private/files/filesystem.php on line 800

[Wed Nov  4 14:33:06 2015] PHP Stack trace:

[Wed Nov  4 14:33:06 2015] PHP   1. Icewind\Streams\CallbackWrapper->stream_close() /home/travis/build/owncloud/core/3rdparty/icewind/streams/src/CallbackWrapper.php:0

[Wed Nov  4 14:33:06 2015] PHP   2. call_user_func:{/home/travis/build/owncloud/core/3rdparty/icewind/streams/src/CallbackWrapper.php:106}() /home/travis/build/owncloud/core/3rdparty/icewind/streams/src/CallbackWrapper.php:106

[Wed Nov  4 14:33:06 2015] PHP   3. OC\Files\View->OC\Files\{closure}() /home/travis/build/owncloud/core/3rdparty/icewind/streams/src/CallbackWrapper.php:106

[Wed Nov  4 14:33:06 2015] PHP   4. OC\Files\View->unlockFile() /home/travis/build/owncloud/core/lib/private/files/view.php:1059

[Wed Nov  4 14:33:06 2015] PHP   5. OC\Files\Filesystem::normalizePath() /home/travis/build/owncloud/core/lib/private/files/view.php:1926

[Wed Nov  4 14:33:06 2015] PHP Fatal error:  Cannot redeclare class RemoteException in /home/travis/build/owncloud/core/remote.php on line 39

[Wed Nov  4 14:33:06 2015] 127.0.0.1:52354 [200]: /remote.php/dav/files/admin/litmus/res-%e2%82%ac - Cannot redeclare class RemoteException in /home/travis/build/owncloud/core/remote.php on line 39

File: apps/dav/appinfo/database.xml [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+		calendarcolor VARBINARY(10),
+		striptodos TINYINT(1) NULL,
+		stripalarms TINYINT(1) NULL,
+		stripattachments TINYINT(1) NULL,
+		lastmodified INT(11) UNSIGNED,
+		UNIQUE(principaluri, uri)
+	) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+	-->
+<table>
+	<name>calendarsubscriptions</name>
```
Comment (phsc) [inline apps/dav/appinfo/database.xml]: *dbprefix* missing?

File: apps/dav/appinfo/database.xml [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+		calendardata MEDIUMBLOB,
+		uri VARBINARY(200),
+		lastmodified INT(11) UNSIGNED,
+		etag VARBINARY(32),
+		size INT(11) UNSIGNED NOT NULL
+	) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+	-->
+
+	<table>
+		<name>schedulingobjects</name>
```
Comment (phsc) [inline apps/dav/appinfo/database.xml]: *dbprefix* missing?

**Comment by phsc** [comment]
It would be very cool, to have a separate CalDAV link for the contacts birthdays. Then all the contacts birthdays would be automatically shown in the calendar.
Maybe something similar like this:
https://github.com/libasys/calendarplus/commit/dd7ce9359f3f885473c5d8271c22528a5cc0dbfe

**Comment by DeepDiver1975** [comment]
continuation: https://github.com/owncloud/core/pull/20371

**Pull request closed without merging**